test/specs/rules/disallow-padding-newlines-in-blocks.js

```diff
@@ -3,58 +3,267 @@ var assert = require('assert');
-    });
-    it('should report extra padding newline before closing brace', function() {
-        assert(checker.checkString('if (true) {\nabc();\n\n}').getErrorCount() === 1);
-    });
-    it('should report extra padding newlines in both cases', function() {
-        assert(checker.checkString('if (true) {\n\nabc();\n\n}').getErrorCount() === 2);
-    });
-    it('should not report with no spaces', function() {
-        assert(checker.checkString('if (true) {\nabc();\n}').isEmpty());
+    trueConfig.forEach(function(config) {
+        describe(config, function() {
+            beforeEach(function() {
+                checker.configure({ disallowPaddingNewlinesInBlocks: config });
+            });
+            it('should not report missing newline after opening brace', function() {
+                assert(checker.checkString('if (true) {abc();\n}').isEmpty());
+            });
+            it('should not report missing newline before closing brace', function() {
+                assert(checker.checkString('if (true) {\nabc();}').isEmpty());
+            });
+            it('should report extra padding newline after opening brace', function() {
+                assert(checker.checkString('if (true) {\n\nabc();\n}').getErrorCount() === 1);
+            });
+            it('should report extra padding newline before closing brace', function() {
+                assert(checker.checkString('if (true) {\nabc();\n\n}').getErrorCount() === 1);
+            });
+            it('should report extra padding newlines in both cases', function() {
+                assert(checker.checkString('if (true) {\n\nabc();\n\n}').getErrorCount() === 2);
+            });
+            it('should not report with no spaces', function() {
+                assert(checker.checkString('if (true) {\nabc();\n}').isEmpty());
+            });
+            it('should not report with no spaces in allman style', function() {
+                assert(checker.checkString('if (true)\n{\nabc();\n}').isEmpty());
+            });
+            it('should not report with comment on first line', function() {
+                assert(checker.checkString('if (true) {\n//comment\nabc();\n}').isEmpty());
+            });
+            it('should not report with multi-line comment on first line', function() {
+                assert(checker.checkString('if (true) {\n/*\ncomment\n*/\nabc();\n}').isEmpty());
+            });
+            it('should not report single line empty function definitions', function() {
+                assert(checker.checkString('var a = function() {};').isEmpty());
+            });
+            it('should not report multiline empty function definitions', function() {
+                assert(checker.checkString('var a = function() {\n};').isEmpty());
+            });
+            it('should report extra padding newline followed by comments', function() {
+                assert(checker.checkString('if (true) {\n\n//comment\n\n/* comments */\n}').getErrorCount() === 1);
```

Comment:
Well, we will start using `chai` "soon" - https://github.com/jscs-dev/node-jscs/pull/1542